subprojects/launcher/src/integTest/groovy/org/gradle/launcher/CommandLineIntegrationSpec.groovy

```diff
@@ -96,26 +96,4 @@ class CommandLineIntegrationSpec extends AbstractIntegrationSpec {
 
         free
     }
-
-    def "cannot combine --scan and --no-scan"() {
-        given:
-        requireGradleDistribution()
-        file("buildSrc/src/main/groovy/BuildScanPlugin.groovy").text = """
-            package com.gradle.test.build.dummy
-            import org.gradle.api.Plugin
-            import org.gradle.api.Project
-
-            class BuildScanPlugin implements Plugin<Project> {
-                void apply(Project project){
-                }
-            }
-        """
-
-        when:
-        args("--scan", "--no-scan")
-
-        then:
-        fails("tasks")
-        errorOutput.contains("Command line switches '--scan' and '--no-scan' are mutually exclusive and must not be used together.")
```

Comment:
Shouldn't this still be caught be the command line parser?

Reply:
`CommandLineParser.allowOneOf(...)` [only picks the last option](https://github.com/gradle/gradle/blob/master/subprojects/cli/src/main/java/org/gradle/cli/CommandLineParser.java#L165) but doesn't throw an exception.

Reply:
And validating that for all options would be breaking, I guess?

Reply:
I don't think it would be breaking necessarily. Probably something we could do anytime at least from my perspective. If you want I can create a new issue for it.